homeassistant/components/arve/entity.py

```diff
@@ -0,0 +1,42 @@
+"""Arve base entity."""
+
+from __future__ import annotations
+
+from asyncarve import Arve, ArveError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.entity import Entity
+
+from .const import LOGGER
+
+
+class ArveDeviceEntity(Entity):
+    """Defines a base Arve device entity."""
+
+    _attr_has_entity_name = True
+    _attr_available = True
+
+    def __init__(self, arve: Arve, entry: ConfigEntry) -> None:
+        """Initialize the Arve device entity."""
+
+        self._entry = entry
+        self.arve = arve
+
+    async def async_update(self) -> None:
```

Comment:
I suggest you move to a `DataUpdateCoordinator`. That way you can get an update for all entities at once (since they all appear to be using the same endpoint), and don't need to do updates per entity.

Reply:
Thank you very much for your suggestion, does it mean that I will need to get rid of the Entity descriptions, or only from value_fn field in this descriptions in sensor.py ?

Reply:
Thanks for the information

Reply:
Hi, I'm trying to implement `DataUpdateCoordinator` as you suggested, but to be honest I'm having difficulties with it. I looked into the documentation that you provided as well as into simple example. I've also found a couple of other examples. In those examples it seems that the Python client takes no place in entities generation, however I would like to have both of the coordinator and Python client in this process. The reason why I would like to do it is because Python client has some attributes that need to be passed into the entities and device itself. And here comes the problem that there is some kind of conflict between the coordinator and Python client. Sharing the screenshot with you: 
<img width="1176" alt="Bildschirmfoto 2024-03-15 um 15 42 01" src="https://github.com/home-assistant/core/assets/146177275/ba21a3b5-4a25-4c52-a804-a7f36067eb51">
Thank you very much for your help! Please let me know if you need the source code and how can I share it with you.

Reply:
You can store the python client inside the coordinator and pass/access it from there. You can just share your GitHub branch if you need help

Reply:
Here is the link to the code: https://github.com/ikalnyi/arve-ha-integration

Reply:
You can get rid of the update methods in `entity.py`, the update will be handled by the coordinator. Its `_async_update_data` is already automatically called, you don't need to call that. Also, you can inherit your base class from the `CoordinatorEntity`. 
There is no point passing the `entry` or `arve` to the base class `init`, as the coordinator already has both.
We can probably also get rid of the `_handle_coordinator_update`, if you utilize a `value_fn` like [here](https://github.com/home-assistant/core/blob/dev/homeassistant/components/lamarzocco/sensor.py). Just let me know if that helps and if you have any further questions.

Reply:
Hello, thank you for help, I've tried the steps above, but the issue still persists. I've updated the public repository, so you can check out the code.

Reply:
tbh, not really sure what's happening. Can you add `await coordinator.async_config_entry_first_refresh()` before adding the coordinator to the `hass.data`. You can also just do `hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator`, no need for the dictionary

Reply:
also probably best you just merge it into this branch, so it's easier to comment on code

Reply:
Thanks, I was able to fix the issue, committed the new code

Reply:
what did the trick? You still have a couple of open items to address ;)

Reply:
Needed to add `super().__init__(coordinator)` line to the the `__init__` function of `ArveDevice` class in `sensor.py`. Also there were a couple of other problems, that were displayed when I've tried to commit changes